### ISSUE TITLE
Missing assets are deserialized as `None`

### ISSUE THREAD

**Issue by gadomski** [stac-utils/stac-geoparquet#77]
If many items are converted to a table, and then converted back to dictionaries, any missing assets are converted to `None` (which is invalid STAC):

```python
import pystac
import stac_geoparquet
from pystac import Item

item: Item = pystac.read_file(
    "https://raw.githubusercontent.com/radiantearth/stac-spec/v1.0.0/examples/simple-item.json"
)
reduced_item = item.full_copy()
del reduced_item.assets["thumbnail"]

table = stac_geoparquet.arrow.parse_stac_items_to_arrow([item, reduced_item])
items = list(stac_geoparquet.arrow.stac_table_to_items(table))
assert items[1]["assets"][
    "thumbnail"
], f"the thumbnail asset is {items[1]['assets']['thumbnail']}"
```

Output:

```text
Traceback (most recent call last):
  File "check.py", line 13, in <module>
    assert items[1]["assets"][
           ^^^^^^^^^^^^^^^^^^^
AssertionError: the thumbnail asset is None
```

**Comment by kylebarron** [comment]
JSON is more descriptive than Arrow around `null` and `undefined`. Because Arrow is columnar, we essentially only preserve `null` and not `undefined` (because the column is defined).

I believe pyarrow serializes all Arrow `null` values as `None` by default. But I agree we shouldn't be able to construct invalid STAC items, so perhaps we should manually remove `None` from asset values? Anywhere else that `None` is invalid? Or should we be coercing `None` to `undefined` everywhere?

**Comment by gadomski** [comment]
> perhaps we should manually remove None from asset values?

Yup, that's what I did [here](https://github.com/stac-utils/stac-rs/blob/a39b13b73975a372d478007ff64b17319defbbd9/stac-arrow/src/json.rs#L440-L442). I ran across the issue when doing some test translations of 1000 sentinel-2 items from the PC — some of the items had a missing preview asset, which is a sort-of-common thing to happen in real-world systems in my experience.

> Anywhere else that None is invalid?

I _think_ in most cases it's ok, and that `assets` is a bit of a special case.

**Comment by gadomski** [comment]
Going to close as completed by https://github.com/stac-utils/stac-geoparquet/pull/111 as well.
